File: ui-web/src/lib/game/battlePass.ts
// TypeScript content with proper types

interface Reward {
  name: string;
  type: 'free' | 'premium';
}

interface BattlePass {
  seasonTheme: string;
  duration: string;
  freeRewards: Reward[];
  premiumRewards: Reward[];
  price: string;
}

export const neonNightAdventures: BattlePass = {
  seasonTheme: 'Neon Night Adventures',
  duration: '8 weeks',
  freeRewards: [
    { name: 'Basic skin', type: 'free' },
    { name: 'Small currency packs', type: 'free' }
  ],
  premiumRewards: [
    { name: 'Exclusive skins', type: 'premium' },
    { name: 'XP boosters', type: 'premium' },
    { name: 'Currency packs', type: 'premium' }
  ],
  price: '$4.99'
};

export function getBattlePassRewards(isPremium: boolean): Reward[] {
  return isPremium ? [...neonNightAdventures.freeRewards, ...neonNightAdventures.premiumRewards] : neonNightAdventures.freeRewards;
}

File: core/internal/game/battle_pass.go
package game

import "time"

// BattlePassConfig holds the configuration for the battle pass
var BattlePassConfig = struct {
	SeasonTheme   string
	Duration      time.Duration
	FreeRewards   []string
	PremiumRewards []string
	Price         string
}{
	SeasonTheme: "Neon Night Adventures",
	Duration:    8 * 7 * 24 * time.Hour, // 8 weeks
	FreeRewards: []string{
		"Basic skin",
		"Small currency packs",
	},
	PremiumRewards: []string{
		"Exclusive skins",
		"XP boosters",
		"Currency packs",
	},
	Price: "$4.99",
}

// GetBattlePassDetails returns the battle pass configuration
type BattlePassDetails struct {
	SeasonTheme   string   `json:"seasonTheme"`
	Duration      string   `json:"duration"`
	FreeRewards   []string `json:"freeRewards"`
	PremiumRewards []string `json:"premiumRewards"`
	Price         string   `json:"price"`
}

func GetBattlePassDetails() BattlePassDetails {
	return BattlePassDetails{
		SeasonTheme:   BattlePassConfig.SeasonTheme,
		Duration:      "8 weeks",
		FreeRewards:   BattlePassConfig.FreeRewards,
		PremiumRewards: BattlePassConfig.PremiumRewards,
		Price:         BattlePassConfig.Price,
	}
}
